Aleksei/extension queue length check

iambeone opened the pull request:
Closes #ISSUE

**Description:**

1. Checking extension queue size before open ActionModal
2. Canceling unsigned transactions in extension on the cancel button pressed

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

faboweb commented on src/ActionModal/components/ActionModal.vue:
```diff
@@ -537,20 +538,35 @@ export default {
         )
         if (confirmResult) {
-          this.session.currrentModalOpen.close()
+          if (!this.queueNotEmpty) {
```
avoid double negation. hard to read. use `queueEmpty` instead.

faboweb commented:
Needs extension release first

faboweb commented:
@iambeone changelog missing

iambeone commented:
@faboweb it's there. The first changed file in Files changed tab

faboweb commented:
Missing coverage still

iambeone commented:
It was adjusted ) Previously this block was removed and now it's back. Line that are missing couldn't be tested or could be just "tested".

faboweb commented:
cover https://codecov.io/gh/luniehq/lunie/compare/15d001d181a32711229e65f13672ab39d6b199ca...69e1607c977bf2aa150ece14c5ddba7cb8128b14/diff#D1-566
cover https://codecov.io/gh/luniehq/lunie/compare/15d001d181a32711229e65f13672ab39d6b199ca...69e1607c977bf2aa150ece14c5ddba7cb8128b14/diff#D7-225

Bitcoinera commented:
I have done some work on this (apollo query coverage). Currently still broken. 

Should I push anyway?

iambeone commented:
I was thinking, i'm going crazy, cause I have fixed this coverage for sure... But the fix is here - https://github.com/luniehq/lunie/pull/3637/files... Updated this pr

Bitcoinera commented:
> Updated this pr

Amazing. Let's merge this then!

iambeone commented:
Don't merge it. Need to do some fixed for identifying missing extension answer

faboweb commented on tests/unit/specs/components/wallet/PageTransactions.spec.js:
```diff
@@ -187,6 +187,65 @@ describe(`PageTransactions`, () => {
+      }
+    )
+
+    expect(result.transactions.length).toBeGreaterThan(0)
```
more important to test the actual content of result.transactions to see if values get created correctly.

faboweb commented on tests/unit/specs/components/wallet/PageTransactions.spec.js:
```diff
@@ -187,6 +187,65 @@ describe(`PageTransactions`, () => {
+      ]
+    })
+
+    expect(result.length).toBeGreaterThan(0)
```
more important to test the actual content of result.transactions to see if values get created correctly.

iambeone replied:
Fixed

Bitcoinera commented on src/ActionModal/components/ActionModal.vue:
```diff
@@ -562,6 +562,7 @@ export default {
       const queue = await this.actionManager.getSignQueue(
         this.selectedSignMethod
       )
+      console.log(queue)
```
Please, don't forget to remove this line

iambeone commented:
Now, it's ok, could be merged

faboweb commented on tests/unit/specs/components/wallet/PageTransactions.spec.js:
```diff
@@ -224,27 +224,28 @@ describe(`PageTransactions`, () => {
-
-    expect(result.length).toBeGreaterThan(0)
+    // adjusting result
+    newTransaction.value = JSON.parse(newTransaction.value)
```
XD

Bitcoinera commented:
Let's merge this! 💪
